lib/gltfPipeline.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+var fs = require('fs');
+var path = require('path');
+var addDefaults = require('./addDefaults');
+var removeUnusedImages = require('../').removeUnusedImages;
+var removeUnusedSamplers = require('../').removeUnusedSamplers;
+var removeUnusedShaders = require('../').removeUnusedShaders;
+var removeUnusedTechniques = require('../').removeUnusedTechniques;
+var removeUnusedPrograms = require('../').removeUnusedPrograms;
+var removeUnusedBuffers = require('../').removeUnusedBuffers;
+var removeUnusedBufferViews = require('../').removeUnusedBufferViews;
+var removeUnusedMaterials = require('../').removeUnusedMaterials;
+var removeUnusedSkins = require('../').removeUnusedSkins;
+var removeUnusedCameras = require('../').removeUnusedCameras;
+var removeUnusedTextures = require('../').removeUnusedTextures;
+var removeUnusedMeshes = require('../').removeUnusedMeshes;
+var removeUnusedNodes = require('../').removeUnusedNodes;
+var removeUnusedAccessors = require('../').removeUnusedAccessors;
+var removeUnused = require('./removeUnused');
+var loadGltfUris = require('./loadGltfUris');
+var parseBinaryGltf = require('./parseBinaryGltf');
+var addPipelineExtras = require('./addPipelineExtras');
+var convertDagToTree = require('./convertDagToTree');
+var combineMeshes = require('./combineMeshes');
+var combinePrimitives = require('./combinePrimitives');
+var removeUnusedVertices = require('./removeUnusedVertices');
+var OptimizationStatistics = require('./OptimizationStatistics');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+var DeveloperError = Cesium.DeveloperError;
+
+
+module.exports = gltfPipeline;
+
+function gltfPipeline(options, callback) {
```

Comment:
Yes, exactly.  At the top-level there should be the command-line app, which just calls a function like this, which then calls a function that takes glTF in-memory as input.

Same deal for output - one function would just provide glTF in-memory and another would write to disk.

So the complete Node.js server -> obj2gltf -> gltf-pipeline flow will not write to disk.

Note that the challenge here is to define the in-memory data structure, i.e., it's one thing for the JSON to be loaded, but what about the .bin, .glsl, and images?  Perhaps gltf-pipeline should document its simple in-memory structure and provide helpers to read and write it (which it already does).